hs_geographic_feature_resource/receivers.py

```diff
@@ -57,13 +60,22 @@ def check_fn_for_shp(files):
                 fileName, fileExtension = os.path.splitext(fullName.lower())
                 if ".shp" == fileExtension:
                     shp_filename = fileName
-                    shp = True
+                    if not shp:
+                        shp = True
+                    else:
+                        return False
```

Comment:
@zhiyuli I got what you are trying to do here, but having a global variable like what you are doing here is confusing to me and make code hard to read. Is it possible to do check for all files all together initially in one function to see whether there is only one shp file in the resource which could eliminate the need for global variables? In general, using global variable is not a good idea and should be avoided if possible.